### ISSUE TITLE
Remove dependency of libboost-python from libeigenpy.sp

### ISSUE THREAD

**Issue by jviereck** [stack-of-tasks/eigenpy#55]
This issue is about supporting python2 and python3 installation of eigenpy.

When compiling and installing `eigenpy`, there are two libraries installed (assuming `prefix=/opt/openrobots` here):

- /opt/openrobots/libs/libeigenpy.so
- /opt/openrobots/lib/python3.5/site-packages/eigenpy.so

Note that the first one has a dependency on the libboost_python lib, in my example

``` bash
$ ldd libeigenpy.so 
	linux-vdso.so.1 =>  (0x00007fffecf99000)
	libboost_python-py35.so.1.58.0 => /usr/lib/x86_64-linux-gnu/libboost_python-py35.so.1.58.0 (0x00007f3e4e3c4000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f3e4e042000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f3e4dd39000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f3e4db23000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f3e4d759000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f3e4d53c000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f3e4e8ca000)
```

(note the second line).

This dependency on `libboost_python` from `libeigenpy.so` creates troubles when trying to install eigenpy for python2 and python3 at the same time in the same folder.

As a workaround, I started to use a `/opt/openrobots` and `/opt/openrobots_py35` folder to separate the two installations of `eigenpy`. However I am wondering if there is a more proper way to resolve this issue and wanted to report this in case someone else comes along the same issue.

**Comment by jcarpent** [comment]
In fact, we should have two separate libraries. One for python2 and the one for python3 in /opt/openrobots/libs. Which means that we just need to add an extended name for each lib according to the version of Python.

**Comment by jcarpent** [comment]
Solved by #57
